### PR TITLE
Add ensembl to refseq transcript id mapping.

### PR DESCRIPTION
**Pull request by bpblanken**

### CODE REVIEW

File: v03_pipeline/lib/annotations/vep.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def vep_110_transcript_consequences_select(
             ),
             c.intron.split('/').map(hl.parse_int32),
         ),
+        refseq_transcript_id=gencode_ensembl_to_refseq_id_mapping.get(c.transcript_id),
```
Comment (bpblanken) [inline v03_pipeline/lib/annotations/vep.py:81]: There's a space saving move here... "set `refseq_transcript_id` to empty if `mane_select` is defined and equal to the result".  It's likely more confusing to the reader code though.

File: v03_pipeline/lib/model/dataset_type.py [hunk 2]
```diff
@@ -214,11 +222,16 @@ def formatting_annotation_fns(
             return GRCh37_fns[self]
         return {
             DatasetType.SNV_INDEL: [
-                *GRCh37_fns[DatasetType.SNV_INDEL],
+                *[
```
Comment (bpblanken) [inline v03_pipeline/lib/model/dataset_type.py:225]: this is our one and only example of an annotation being "overridden" for 38.

File: v03_pipeline/lib/reference_data/gencode/mapping_gene_ids.py [hunk 3]
```diff
@@ -129,3 +132,16 @@ def load_gencode(gencode_release, download_path=''):
...
+    url = GENCODE_ENSEMBL_TO_REFSEQ_URL.format(gencode_release=gencode_release)
+    response = requests.get(url, stream=True)
+    ensembl_to_refseq_ids = {}
+    for line in gzip.GzipFile(fileobj=response.raw):
+        line = line.decode('ascii').strip().split('\t')
+        if len(line) > 3:
+            raise ValueError(
+                'Unexpected number of fields on line in ensemble_to_refseq mapping',
+            )
+        ensembl_to_refseq_ids[line[0].split('.')[0]] = line[1]
```
Comment (bpblanken) [inline v03_pipeline/lib/reference_data/gencode/mapping_gene_ids.py:146]: The mapping structure mirrors 'VEP': no minor version for ensembl, included minor version for refseq.

File: v03_pipeline/lib/annotations/vep.py (hunk 1 above)
Comment (bpblanken) [reply v03_pipeline/lib/annotations/vep.py:81]: decided against this.

**Review by hanars — APPROVED** [review]

**Pull request merged**